### PR TITLE
10 add visual sense to enemies

### LINKED ISSUE #10: Add visual sense to enemies
Enemies should follow the player based on the last seen position such that the player can hide.
- follow based on last position
- hide behind objects

### PR DESCRIPTION
**Pull request by AnomahR**
closes #10

### CODE REVIEW

File: enemy/Enemy.gd [hunk 1]
```diff
@@ -3,16 +3,28 @@ extends KinematicBody2D
...
 	pass
 	
-func _process(delta):
+func _physics_process(delta):
+	$Eyesight.scan(rotation)
 	if follow_target != null:
 		var direction = (follow_target.position - position).normalized()
 		move_and_slide(SPEED * direction)
 
+#func look_for_target(target):
```
Comment (MrGreenTea) [inline enemy/Enemy.gd:18]: please delete commented code

File: enemy/Eyesight.gd [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			prints("Found player", player)
+	
+	# add collision exception for tiles
+	for node in get_tree().get_nodes_in_group("invisible_raycast"):
+		add_exception(node)
+		prints("Added exception", node)
+	
+func scan(scan_angle):
+	visible_things = []
+	#for angle in range(scan_angle-eyesight_angle_radians/2, scan_angle+eyesight_angle_radians/2, eyesight_angle_steps):
```
Comment (MrGreenTea) [inline enemy/Eyesight.gd:30]: please delete commented code

File: enemy/Eyesight.gd [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			player = node
+			prints("Found player", player)
+	
+	# add collision exception for tiles
+	for node in get_tree().get_nodes_in_group("invisible_raycast"):
+		add_exception(node)
+		prints("Added exception", node)
+	
+func scan(scan_angle):
+	visible_things = []
```
Comment (MrGreenTea) [inline enemy/Eyesight.gd:29]: this variable is unused, right?

**Comment by AnomahR** [comment]
New commit!

**Pull request merged**